File: mizo/app/src/main/java/com/example/mizo/MainActivity.java
package com.example.mizo;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.airbnb.lottie.LottieAnimationView;

public class MainActivity extends AppCompatActivity {

    LottieAnimationView lottie;
    Button start;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        start=findViewById(R.id.start);
        lottie = findViewById(R.id.meal);

        lottie.animate().setDuration(10000).setStartDelay(2900);

        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),MainActivity2.class);
                startActivity(i);
            }
        });
    }
}

File: day3_BoundActivity/app/src/main/java/com/example/day3_boundactivity/MainActivity.kt
package com.example.day3_boundactivity

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    lateinit var dateandtime:TextView
    lateinit var getDate:Button
    lateinit var myService:MyBoundService
    var isBound:Boolean = false
    lateinit var mydate :String
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        dateandtime = findViewById(R.id.tv_date)
        getDate = findViewById(R.id.button)


        val intent = Intent(this,MyBoundService::class.java)
        bindService(intent,myConniction,Context.BIND_AUTO_CREATE)


        getDate.setOnClickListener {
            mydate = myService.getTime()
            dateandtime.text = mydate
        }

    }


    private var myConniction : ServiceConnection = object : ServiceConnection{
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
          val binder : MyBoundService.MyBinder = service as MyBoundService.MyBinder
           myService=binder.getBiundService()
            isBound=true
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            isBound=false
        }
    }
}